framework/src/preconditioners/FieldSplitPreconditioner.C

```diff
@@ -60,22 +53,24 @@ FieldSplitPreconditioner::FieldSplitPreconditioner(const InputParameters & param
   // it is recommended to have a full Jacobian for using
   // the fieldSplit preconditioner
   bool full = getParam<bool>("full");
+
   // how variables couple
   std::unique_ptr<CouplingMatrix> cm = std::make_unique<CouplingMatrix>(n_vars);
   if (!full)
   {
+    const auto off_diag_rows = getParam<std::vector<NonlinearVariableName>>("off_diag_row");
+    const auto off_diag_columns = getParam<std::vector<NonlinearVariableName>>("off_diag_column");
```

Comment:
Is this enforcing both parameters to be of equal length? (probably in the hidden code). If so then a better way would be 

```
const auto off_diag = getParam<NonlinearVariableName, NonlinearVariableName>("off_diag_row", "off_diag_column");
```

to get a vector of pairs (with built in length error checking. Then you can simplify teh code below to 

```
for (const auto & item: off_diag)
{
  auto row = _nl.getVariable(0, item.first).number();
  auto column = _nl.getVariable(0, item.second).number();
  (*cm)(row, column) = 1;
}
```